### ISSUE TITLE
[Question] Add multiple firmware to the same device profile

### ISSUE THREAD

**Issue by pintify** [thingsboard/thingsboard#9962]
**Component**

 * Generic

**Description**

I need to integrate Thingsboard with a collection of devices who run multiple services (some of the dockerized, some of the independent) and such services may be updated independently one of another. I'm currently working on the MQTT client process that will run in these devices to manage the configuration of these services and the upgrade of them, but apparently the MQTT API only allows one firmware to be assigned at a time to the device. How could I hold this scenario so that the update of these services can be done independently and even let multiple of them scheduled?

**Comment by mykhailokornieiev** [comment]
This question is similar to https://github.com/thingsboard/thingsboard/issues/9905 and unfortunatley, there is no possibility to assign more than one FW and one SW per device / device profile

**Comment by pintify** [comment]
Alright, thank you very much. We'll try to find a workaround, then.

**Comment by davidpob99** [comment]
@pintify did you find a workaround?

**Comment by pintify** [comment]
Indeed, but you may not like it. At the moment, we use a client attribute to provide the list of installed packages and versions (or however you call it: containers, software, plugins). This way the platform knows what the device has. The install/update and uninstall instructions are custom client RPCs.
